robotiq_force_torque_sensor/nodes/rq_sensor.cpp

```diff
@@ -171,206 +297,49 @@ static robotiq_force_torque_sensor::ft_sensor get_data(void)
-        sensor_pub = nh.advertise<robotiq_force_torque_sensor::ft_sensor>("robotiq_force_torque_sensor", 512);
-        wrench_pub = nh.advertise<geometry_msgs::WrenchStamped>("robotiq_force_torque_wrench", 512);
-        watchdog_pub = nh.advertise<std_msgs::Empty>("/robotiq_ft300_force_torque_sensor/watchdog", 1);
-        service = nh.advertiseService("robotiq_force_torque_sensor_acc", receiverCallback);
-        created_publishers = 1;
+    else {
+        ROS_INFO("Trying to Start Robotiq FT300 Sensor");
+        sensor_ptr.reset(new RealSensor);
     }
 
-    //std_msgs::String msg;
-    geometry_msgs::WrenchStamped wrenchMsg;
-    std_msgs::Empty watchdog_msg;
-    ros::param::param<std::string>("~frame_id", wrenchMsg.header.frame_id, "robotiq_force_torque_frame_id");
+    uint32_t publish_count = 0;
 
-    // Publish the connection state
-    connection_msg.data = connected;
-    connection_pub.publish(connection_msg);
-    ros::spinOnce();
-
-    if (connected)
-    {
-        ROS_INFO("Starting Sensor");
+    sensor_ptr->Connect();
+    if (!sensor_ptr->isConnected()) {
+        launch_control_link.send_failure();
+        return -1;
     }
+    launch_control_link.send_normal();
 
-    while(ros::ok() && connected)
-    {
-        connected = try_connection();
-
-        if(rq_sensor_get_current_state() == RQ_STATE_RUN)
-        {
-            strcpy(bufStream,"");
-            msgStream = get_data();
-
-            if(rq_state_got_new_message())
-            {
-                sensor_pub.publish(msgStream);
-
-                // compose WrenchStamped Msg
-                wrenchMsg.header.stamp = ros::Time::now();
-                wrenchMsg.wrench.force.x = msgStream.Fx;
-                wrenchMsg.wrench.force.y = msgStream.Fy;
-                wrenchMsg.wrench.force.z = msgStream.Fz;
-                wrenchMsg.wrench.torque.x = msgStream.Mx;
-                wrenchMsg.wrench.torque.y = msgStream.My;
-                wrenchMsg.wrench.torque.z = msgStream.Mz;
-                wrench_pub.publish(wrenchMsg);
-            }
-        }
+    while (ros::ok() && sensor_ptr->isConnected()) {
+        sensor_ptr->publishSensor();
 
-        // Publish the connection state every 1000 iterations
-        if (publish_count % 10 == 0)
-        {
-            watchdog_pub.publish(watchdog_msg);
+        if (publish_count % 10 == 0) {
+            sensor_ptr->publishWatchdog();
         }
-
-        if (publish_count == 1000)
-        {
-            connection_msg.data = connected;
-            connection_pub.publish(connection_msg);
+        if (publish_count == 1000) {
+            sensor_ptr->publishConnection();
             publish_count = 0;
         }
         publish_count++;
-        ros::spinOnce();
     }
 
-    if (!connected)
-    {
-        connection_msg.data = connected;
-        connection_pub.publish(connection_msg);
-        ros::spinOnce();
-    }
-
-    ROS_INFO("Sensor Stopped");
-    if (created_publishers)
```

Comment:
Please leave this in unless testing proves that it is not required when undergoing rapid node restarts.

Reply:
Implementation of  publishers issue shutdown on destruction, so should be redundant.